packages/http/lib/auth.tsp

```diff
@@ -105,14 +108,18 @@ model ApiKeyAuth<Location extends ApiKeyLocation, Name extends string> {
  * For more information about OAuth 2.0, see oauth.net and RFC 6749.
  *
  * @template Flows The list of supported OAuth2 flows
+ * @template Scopes The list of OAuth2 scopes, which are common for every flow from `Flows`. This list is combined with the scopes defined in specific OAuth2 flows.
  */
 @doc("")
-model OAuth2Auth<Flows extends OAuth2Flow[]> {
+model OAuth2Auth<Flows extends OAuth2Flow[], Scopes extends string[] = []> {
   @doc("OAuth2 authentication")
   type: AuthType.oauth2;
 
   @doc("Supported OAuth2 flows")
   flows: Flows;
+
+  @doc("Oauth2 scopes, which are common for every flow")
+  commonScopes: Scopes;
```

Comment:
I wonder if using `commonScopes` here is the right choice vs `defautlScopes` for all the flows. Problem being that if a flow has a different set of scopes you then you now have to omit commonScopes completely and pass the scopes to each flow instead of just overriding it for one.
Having an append pattern might seems like a less verbose solution but it can get us stuck in the future.

Reply:
Fixed: https://github.com/microsoft/typespec/pull/2901/commits/29fa3dea43082e5383bdc2ce8ffac0cac12ead72